solutions/advanced/Ligatures.mdx

```diff
@@ -0,0 +1,272 @@
+---
+id: Ligatures
+source: Kattis
+title: Ligatures
+author: Alexis Tao
+---
+
+### Explanation
+
+The problem involves counting the occurrences of specific pairs of characters (ligatures) in a given text corpus based on multiple sets of suggested ligatures. The primary challenge is efficiently handling large input sizes and ensuring that the counting process respects the rule that ligatures can't overlap.
+
+### Solution
+
+This solution to the problem uses SIMD (Single Instruction, Multiple Data) parallelism to efficiently count the occurrences of suggested ligatures within a large corpus:
+
+**Overview**
+1. **Preprocessing the Corpus**:
+- The corpus is preprocessed to identify all pairs of consecutive characters, which are stored in a pairs vector.
+
+2. **Setting Up Bitmasks**:
+- A bitmask array active is set up where each bit represents whether a specific ligature is part of a query.
+The bitmask for each query is configured such that each suggested ligature toggles a specific bit.
+
+3. **Counting Occurrences Using SIMD**:
+- The core of the solution utilizes SIMD instructions to parallelize the counting process.
+The program processes the corpus in chunks, accumulating counts for each query simultaneously using AVX2 intrinsics.
+
```

Comment:
add `<IncompleteSection />` around here given that this solution explanation is by no means complete.